scipy/optimize/tests/test__linprog_clean_inputs.py

```diff
@@ -237,33 +240,72 @@ def test__clean_inputs3():
 def test_bad_bounds():
     lp = _LPProblem(c=[1, 2])
 
-    assert_raises(ValueError, _clean_inputs, lp._replace(bounds=(1, -2)))
-    assert_raises(ValueError, _clean_inputs, lp._replace(bounds=[(1, -2)]))
-    assert_raises(ValueError, _clean_inputs, lp._replace(bounds=[(1, -2), (1, 2)]))
-
     assert_raises(ValueError, _clean_inputs, lp._replace(bounds=(1, 2, 2)))
     assert_raises(ValueError, _clean_inputs, lp._replace(bounds=[(1, 2, 2)]))
     assert_raises(ValueError, _clean_inputs, lp._replace(bounds=[(1, 2), (1, 2, 2)]))
     assert_raises(ValueError, _clean_inputs, lp._replace(bounds=[(1, 2), (1, 2), (1, 2)]))
 
+    lp = _LPProblem(c=[1, 2, 3, 4])
+
+    assert_raises(ValueError, _clean_inputs, lp._replace(bounds=[(1, 2, 3, 4), (1, 2, 3, 4)]))
+
 
 def test_good_bounds():
     lp = _LPProblem(c=[1, 2])
 
     lp_cleaned = _clean_inputs(lp)  # lp.bounds is None by default
-    assert_(lp_cleaned.bounds == [(0, None)] * 2, "")
+    assert_(np.all(lp_cleaned.bounds == [(0, np.inf)] * 2), "")
 
     lp_cleaned = _clean_inputs(lp._replace(bounds=(1, 2)))
-    assert_(lp_cleaned.bounds == [(1, 2)] * 2, "")
+    assert_(np.all(lp_cleaned.bounds == [(1, 2)] * 2), "")
```

Comment:
For these places where you're wrapping the comparison in `np.all`, I agree that it's the right thing to do. I'm curious, though - did the changes you make to the code require that you change the tests like this in order to pass?

Reply:
Yes, it raises a ValueError: `ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`.

Reply:
Oh, I was wondering why they weren't failing before - but I realize it was because they were lists, not arrays.